Mathlib/LinearAlgebra/BilinearMap.lean

```diff
@@ -208,6 +208,43 @@ theorem domRestrict₁₂_apply (f : M →ₛₗ[ρ₁₂] N →ₛₗ[σ₁₂]
     (x : p) (y : q) : f.domRestrict₁₂ p q x y = f x y := rfl
 #align linear_map.dom_restrict₁₂_apply LinearMap.domRestrict₁₂_apply
 
+section restrictScalars
+
+variable (R' S' : Type _)
+
+variable [Semiring R'] [Semiring S'] [Module R' M] [Module S' N] [Module R' Pₗ] [Module S' Pₗ]
+
+variable [SMulCommClass S' R' Pₗ] [SMulCommClass S R' Pₗ]
+
+variable [SMul S' S] [IsScalarTower S' S N] [IsScalarTower S' S Pₗ]
+
+variable [SMul R' R] [IsScalarTower R' R M] [IsScalarTower R' R Pₗ]
+
+/-- If `B : M → N → Pₗ` is `R`-`S` bilinear and `R'` and `S'` are compatible scalar multiplications,
+then the restriction of scalars is a `R'`-`S'` bilinear map.-/
+def restrictScalars₂ (B : M →ₗ[R] N →ₗ[S] Pₗ) : M →ₗ[R'] N →ₗ[S'] Pₗ :=
+  LinearMap.mk₂' R' S'
+    (fun x y ↦ B x y)
+    (by intros; simp only [map_add, LinearMap.add_apply])
+    (by intros; simp only [LinearMap.map_smul_of_tower, LinearMap.smul_apply])
+    (fun x ↦ map_add (B x))
+    (fun _ x ↦ (B x).map_smul_of_tower _)
```

Comment:
I assume my previous `(f.flip.restrictScalars _).flip.restrictScalars _` implementation no longer works?

Reply:
No, this requires weird instances that totally unnecessary.